src/app/edit.component/edit.component.html

```diff
@@ -0,0 +1,6 @@
+
+<div >
+  <input type="name"  class="form-control" required
+  [(ngModel)]="todo.name" >
+  <button *ngIf="edit===true" (click)="editdata(todo)">update</button>
```

Comment:
editdata(todo) what is the use of this function because it is not defined in its component.

Reply:
sir it is define and use in edit.component.ts file
editdata(selected: Todo) {
        this._HeroService.editData(selected);
        this._router.navigate(['Dashboard']);
    }